### PR TITLE
Make Gradle client start up faster

### PR DESCRIPTION
**Pull request by oehme**
This is a set of small improvements aimed at making the Gradle CLI start up faster. The biggest change is getting rid of a bunch of unused services when the actual build is run inside the daemon. See the individual commits for more explanations.

On my machine this reduces CLI startup from 750ms to 530ms. The difference is a bit smaller on CI, which is probably because it uses Linux and I'm on MacOS. Linux is faster at I/O, so the I/O-reducing changes have less of an effect. The effect on Windows should thus be even bigger.

### CODE REVIEW

File: subprojects/core/src/main/java/org/gradle/process/internal/JvmOptions.java [hunk 1]
```diff
@@ -307,7 +314,9 @@ public void copyTo(JavaForkOptions target) {
         target.setSystemProperties(mutableSystemProperties);
         target.setMinHeapSize(minHeapSize);
         target.setMaxHeapSize(maxHeapSize);
-        target.setBootstrapClasspath(bootstrapClasspath);
+        if (bootstrapClasspath != null) {
+            target.setBootstrapClasspath(bootstrapClasspath);
+        }
```
Comment (adammurdoch) [inline subprojects/core/src/main/java/org/gradle/process/internal/JvmOptions.java]: probably should be: else set empty bootstrap classpath on target.

File: subprojects/native/src/main/java/org/gradle/internal/nativeintegration/filesystem/services/GenericFileSystem.java [hunk 2]
```diff
@@ -96,16 +97,22 @@ public GenericFileSystem(FileModeMutator chmod, FileModeAccessor stat, Symlink s
...
-            file = createFile(content);
-            caseSensitive = probeCaseSensitive(file, content);
-        } catch (IOException e) {
-            throw new RuntimeException(e);
-        } finally {
-            FileUtils.deleteQuietly(file);
+    }
+
+    private void initializeCaseSensitive() {
+        if (caseSensitive == null) {
```
Comment (adammurdoch) [inline subprojects/native/src/main/java/org/gradle/internal/nativeintegration/filesystem/services/GenericFileSystem.java]: thread safety?

Comment (oehme) [inline subprojects/native/src/main/java/org/gradle/internal/nativeintegration/filesystem/services/GenericFileSystem.java]: The worst thing that could happen is that two threads do this work twice. They'll agree on the same answer in the end though.

File: subprojects/core/src/main/java/org/gradle/process/internal/JvmOptions.java (hunk 1 above)
Comment (oehme) [inline subprojects/core/src/main/java/org/gradle/process/internal/JvmOptions.java]: Fixed. Also fixed a similar error introduced earlier in another class.

File: subprojects/native/src/main/java/org/gradle/internal/nativeintegration/filesystem/services/GenericFileSystem.java (hunk 2 above)
Comment (adammurdoch) [inline subprojects/native/src/main/java/org/gradle/internal/nativeintegration/filesystem/services/GenericFileSystem.java]: I don't like this approach in general, as it's easy to overlook this when making changes. There should be _some_ kind of explicit declaration that this is and needs to be thread safe. Which could be a comment or an annotation or some code or something else.

Comment (oehme) [inline subprojects/native/src/main/java/org/gradle/internal/nativeintegration/filesystem/services/GenericFileSystem.java]: Sure, I can document that.

**Pull request merged**